src/Symfony/Component/Workflow/Workflow.php

```diff
@@ -254,17 +256,16 @@ private function transition($subject, Transition $transition, Marking $marking)
 
     private function enter($subject, Transition $transition, Marking $marking)
     {
+        foreach ($places = $transition->getTos() as $place) {
+            $marking->mark($place);
```

Comment:
This is a bit weird to me. The event is named `enter` and not `entered`. So for me the event should be dispatched **before** the actual marking.